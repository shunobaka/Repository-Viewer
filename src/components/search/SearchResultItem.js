import React from 'react';
import { Link } from 'react-router-dom';
import { Col, Image, Jumbotron } from 'react-bootstrap';
import PropTypes from 'prop-types';

const UserItem = ({ user }) => {
  return (
    <Col md={3}>
      <Jumbotron>
        <Link to={`/user/${user.username}`}>
          <h4 className="text-center">{user.username}</h4>
          <Image width="100%" src={user.avatar} />
        </Link>
      </Jumbotron>
    </Col>
  );
};

UserItem.propTypes = {
  user: PropTypes.shape({
    login: PropTypes.string,
    avatar_url: PropTypes.string,
  }).isRequired,
};

export default UserItem;
