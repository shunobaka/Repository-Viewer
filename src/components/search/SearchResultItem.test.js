import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SearchResultItem from './SearchResultItem';

describe('searchResultItem snapshots', () => {
  const mockUser = {
    username: 'Test',
  };

  it('mock user renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SearchResultItem user={mockUser} />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
