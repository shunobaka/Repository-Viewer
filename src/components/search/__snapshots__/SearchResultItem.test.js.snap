// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchResultItem snapshots mock user renders correctly 1`] = `
<div
  className="col-md-3"
>
  <div
    className="jumbotron"
  >
    <a
      href="/user/Test"
      onClick={[Function]}
    >
      <h4
        className="text-center"
      >
        Test
      </h4>
      <img
        className=""
        width="100%"
      />
    </a>
  </div>
</div>
`;
